packages/router-vite-plugin/src/index.ts

```diff
@@ -17,23 +17,32 @@ export function TanStackRouterVite(inlineConfig: Partial<Config> = {}): Plugin {
     }
   }
 
+  const handleFile = async (file: string) => {
+    const filePath = normalize(file)
+    if (filePath === join(ROOT, CONFIG_FILE_NAME)) {
+      userConfig = await getConfig(inlineConfig, ROOT)
+      return
+    }
+    const routesDirectoryPath = isAbsolute(userConfig.routesDirectory)
+      ? userConfig.routesDirectory
+      : join(ROOT, userConfig.routesDirectory)
+    if (filePath.startsWith(routesDirectoryPath)) {
+      await generate()
+    }
+  }
+
   return {
     name: 'vite-plugin-tanstack-router',
     configResolved: async () => {
       userConfig = await getConfig(inlineConfig, ROOT)
       await generate()
     },
     handleHotUpdate: async ({ file }) => {
```

Comment:
Should we still be doing this?

The jsdoc comment for it states:
```
Perform custom handling of HMR updates. The handler receives a context containing changed filename, timestamp, a list of modules affected by the file change, and the dev server instance.

- The hook can return a filtered list of modules to narrow down the update. e.g. for a Vue SFC, we can narrow down the part to update by comparing the descriptors.

- The hook can also return an empty array and then perform custom updates by sending a custom hmr payload via server.ws.send().

- If the hook doesn't return a value, the hmr update will be performed as normal.
```

Reply:
Have you tried using it with this removed? I am just as unsure as you what would happen. But, if we remove it, and it does nothing, then I think it’s the right choice. if you are relatively sure that it will have no side effects, then let’s just remove it and deal with any repercussions that come up. It’s worth the risk.

Reply:
With all the local tests I've done (with it removed), I couldn't find a scenario in which it broke any existing functionality both during `dev` and `build`.
I'll push this today and keep an eye on the support channels, given that the change is so small, reverting this one shouldn't take more than a couple of minutes.